Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2210,6 +2235,33 @@ def visit_ClassDefNode(self, node):
 
 class AnalyseExpressionsTransform(CythonTransform):
 
+    def __init__(self, context):
+        super(AnalyseExpressionsTransform, self).__init__(context)
+        self.parallel_with_block = None
+
+    def visit_SingleAssignmentNode(self, node):
```

Comment:
This seems like it might fit better in MarkParallelAssignments?

Reply:
I remember I tried (this or something else) in an earlier stage but some information needed was not there yet. I will check if it can be done in MarkParallelAssignments.